torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -729,6 +755,104 @@ void Reducer::finalize_backward() {
   local_used_maps_reduced_ = false;
 }
 
+void Reducer::sync_bucket_indices(
+    std::vector<std::vector<size_t>>& bucket_indices) {
+  int64_t broadcast_bucket_size = DEFAULT_BROADCAST_BUCKET_BYTES;
+
+  auto num_buckets = bucket_indices.size();
+  std::vector<size_t> bucket_sizes;
+  int64_t total_size = 0;
+  for (size_t i = 0; i < num_buckets; i++) {
+    auto bucket_size = bucket_indices.at(i).size();
+    bucket_sizes.push_back(bucket_size);
+    total_size += bucket_size;
+  }
+
+  at::TensorOptions options;
+  options = options.dtype(at::kInt);
+  options = options.device(replicas_[0][0].device());
+
+  // Group indices and num_bucket together into indices_tensor
+  // Broadcast this tensor first, as its size is equal among all processes
+  auto indices_tensor = at::empty({total_size + 1}, at::kInt);
+  auto indices_accessor = indices_tensor.accessor<int, 1>();
```

Comment:
Question, do we have to use the accessor here? I recall hearing some update from C++ team that C++ tensor indexing support is ready?

Reply:
I was recommended to use accessor for efficiency last year when I worked on data loading API. I did not find C++ tensor indexing documentation, is it a new feature?